### DISCUSSION TITLE
Zoom-level for POI-overlays

### DISCUSSION THREAD

**Discussion by ThomasMeister123** [osmandapp/OsmAnd#13962] [Q&A] [answered]
Is there a chance to select a zoom-level for POI-overlays? So that the POI are only visible, when I zoom to details

![Screenshot_20220321-193625](https://user-images.githubusercontent.com/62500258/159341344-bdab25c9-bed3-4eed-9923-4e69eabc3e33.png)

**Comment by Lee-Carre** [comment]
The idea for PoI overlay is to display selected PoIs at any zoom level.

Without the overlay layer, then the regular map will show smaller features at greater zoom levels, and fewer when zoomed out. This seems to be what you're seeking.

Likely the same scenario as #12552.

**Reply by ThomasMeister123** [reply]
Thank you. Thats not what I'm looking for. In this case its the Wikipedia-Overlays. They are not visible in the "normal" map. Or when I select a special POI-type (e.g. just Shell-petrol-stations) then it is also not useful to search these on the standard map

**Reply by Lee-Carre** [reply]
Actually, since this has occurred multiple times, I'm wondering if there's some improvement to the UI which could be made.

@ThomasMeister123 are you able to articulate your thought process behind why you enabled the PoI overlay?

In the (other) cited discussion, it seemed that the user simply wanted PoIs displayed at all, in the normal fashion, and so likely enabled the overlay with all types, not realising what it actually does, and the behaviour he wanted was the default standard.

If so, then something is either _not_ being communicated to users through the UI's design, else something is being _mis_communicated.

**Reply by Lee-Carre** [reply]
> what I'm looking for. In this case its the Wikipedia-Overlays. They are not visible in the "normal" map.

Ah, yes. That is different. I hadn't noticed the type of PoI being mostly Wikipedia, in your screenshot.
I don't know, in that case. The only workaround coming to mind is to toggle them on/off as needed.

The idea being similar to (other) PoIs; highlighting them (at any zoom) in order to emphasise them. Without that, how would a user identify notable places (e.g. to visit, probably in an unfamiliar area)? In some areas, the density is sparse / low, so seeing all of them while far zoomed-out is needed.

Perhaps grouping / clustering would address your concern? That is, when zoomed-out, instead of displaying each individual PoI marker, display an icon with a numeral of how many (currently hidden) PoIs it represents. Tapping on such an icon expands the group / cluster to show each individual PoI.

> Or when I select a special POI-type (e.g. just Shell-petrol-stations) then it is also not useful to search these on the standard map

If I understand correctly, then it sounds like a custom search is what you're after, for that kind of specific (tag, in OSM lingo) matching.

**Reply by ThomasMeister123** [reply]
One example in my case for my "car-map": show POI for fastfood and Touristic attraction :) An other is, like I told above, to see where are Wikipedia-POIs. But I only want see this when, maybe, I zoom to a 30km-tile. When zoom to whole Germany I dont want see 50.000 Wikipedia-Poi-icons

**Comment by ThomasMeister123** [comment, marked as answer]
Not the best, but ONE solution maybe grouping like LocusMaps does it. But select a zoomlevel for showing pois would be better

![Screenshot_20220321-204419](https://user-images.githubusercontent.com/62500258/159352085-dbacc67c-42e4-4f94-aa1a-aac9befec6e1.png)

![Uploading Screenshot_20220321-204419.png…]()

**Reply by Lee-Carre** [reply]
😄 otherwise known as clustering:
> Perhaps grouping / clustering would address your concern? That is, when zoomed-out, instead of displaying each individual PoI marker, display an icon with a numeral of how many (currently hidden) PoIs it represents. Tapping on such an icon expands the group / cluster to show each individual PoI.

👍🙂

I'll take that as a ‘yes’. Happy-accident for the win.

In that case, would be worth opening a regular ticket (issue) to propose this feature.
